pallets/mint_with_fee/src/lib.rs

```diff
@@ -0,0 +1,203 @@
+		fn default() -> Self {
+			GenesisConfig { fee_percent: 10u32.into() }
+		}
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+		fn build(&self) {
+			FeePercent::<T>::put(self.fee_percent);
+		}
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+        /// Mints the given amount of value on the target account,
+        /// and mint a percent of the amount on the fee account, if provided
+        ///
+        /// `mint` will increase the total issuance, and increase the amounts of the targets accounts.
+        ///
+        /// The dispatch origin for this call must be `Signed` by the root.
+        ///
+        /// # <weight>
+        ///
+        /// Related functions:
+        /// - `mint_to_account` can be one or two times, dpending on if the fee account is provided or not.
+        ///
+        ///
+        /// # </weight>
+		#[pallet::weight(T::WeightInfo::mint())]
+        // #[pallet::weight(0)]
+		pub fn mint(
+			origin: OriginFor<T>,
+			target_account: T::AccountId,
+			fee_target_account: Option<T::AccountId>,
+			#[pallet::compact] amount: BalanceOf<T>,
+			metadata: Vec<u8>,
+		) -> DispatchResult {
+			ensure_root(origin)?;
+
+            if metadata.len() > 100 {
+                return Err(Error::<T>::TooLongMetadata.into());
+            }
+
+            if let Some(fee_target_account) = fee_target_account {
+			    let fee_amount = (amount / 100u32.into()).checked_mul(&Self::fee_percent()).ok_or(Error::<T>::Overflow)?;
+
+			    Self::mint_to_account(&fee_target_account, fee_amount)?;
+
+			    Self::deposit_event(Event::<T>::FeeMinted(
+                    fee_target_account,
+                    fee_amount,
+                    metadata.clone(),
+                ));
+            };
+
+			Self::mint_to_account(&target_account, amount)?;
+
+			Self::deposit_event(Event::<T>::ValueMinted(
+				target_account,
+				amount,
+				metadata,
+			));
+
+			Ok(())
+		}
+
+        /// Change the value of the fee percentage in storage
+        ///
+        /// `set_fee` will change the value of the fee percentage in storage,
+        /// affecting the next calls to `mint`
+        ///
+        /// The dispatch origin for this call must be `Signed` by the root.
+		// #[pallet::weight(10_000)]
+		#[pallet::weight(T::WeightInfo::change_fee_percent())]
+		pub fn change_fee_percent(origin: OriginFor<T>, percentage: BalanceOf<T>) -> DispatchResult {
+			ensure_root(origin)?;
+			FeePercent::<T>::put(percentage);
+
+            Self::deposit_event(Event::<T>::FeeChanged(percentage));
+			Ok(())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+        /// Mints the given amount on the target account
+        ///
+        /// cannot fail
+		fn mint_to_account(target_account: &T::AccountId, amount: BalanceOf<T>) -> Result<(), Error<T>> {
+			let negative_amount_imbalance = T::Currency::issue(amount);
+			T::Currency::resolve_creating(&target_account, negative_amount_imbalance);
+
+			Ok(())
+		}
```

Comment:
If it cannot fail, don't have it return an `Result`
No return seems more appropriate there

Reply:
true